src/Components/Reports/Util.test.js

```diff
@@ -0,0 +1,113 @@
+import { generateFilterText, responseToPDFData } from './Util';
+
+describe('Util generateFilterText method tests', () => {
+    it('should generate filter text for name filter', () => {
+        const expectedFilterText = `\nFilters applied\nName: ros-system`;
+        const filters = {
+            hostnameOrId: 'ros-system'
+        };
+        const actualFilterText = generateFilterText(filters);
+
+        expect(actualFilterText).toBe(expectedFilterText);
+    });
+
+    it('should generate filter text for state filter', () => {
+        const expectedFilterText = `\nFilters applied\nState: Undersized,Waiting for data,Oversized\t\t\t\t\t`;
+        const filters = {
+            stateFilter: ['Undersized', 'Waiting for data', 'Oversized']
+        };
+        const actualFilterText = generateFilterText(filters);
+
+        expect(actualFilterText).toBe(expectedFilterText);
+    });
+
+    it('should generate filter text for name & state filter', () => {
+        const expectedFilterText = `\nFilters applied\nState: Undersized,Waiting for data,Oversized\t\t\t\t\tName: ros-system`;
+        const filters = {
+            hostnameOrId: 'ros-system',
+            stateFilter: ['Undersized', 'Waiting for data', 'Oversized']
+        };
+        const actualFilterText = generateFilterText(filters);
+
+        expect(actualFilterText).toBe(expectedFilterText);
+    });
+
+    it('should generate empty filter string in case of no filters', () => {
+        const expectedFilterText = '';
+        const filters = {};
+        const actualFilterText = generateFilterText(filters);
+
+        expect(actualFilterText).toBe(expectedFilterText);
+    });
+
+    it('should generate empty filter string in case of no filters', () => {
+        const expectedFilterText = '';
+        const filters = {
+            hostnameOrId: '',
+            stateFilter: []
+        };
+        const actualFilterText = generateFilterText(filters);
+
+        expect(actualFilterText).toBe(expectedFilterText);
+    });
+
+    it('should generate empty filter string in case of no filters', () => {
+        const expectedFilterText = '';
+        const filters = {
+            hostnameOrId: undefined,
+            stateFilter: []
+        };
+        const actualFilterText = generateFilterText(filters);
+
+        expect(actualFilterText).toBe(expectedFilterText);
+    });
+});
+
+describe('Util responseToPDFData method tests', () => {
+    it('should generate array of data in the format required to generate PDF', () => {
+        const expectedSystemsRowsData = [
+            ['ip-172-31-28-69.ec2.internal', 'RHEL 8.4', '90%', '97%', '0.314', '1', 'Undersized', '03 Mar 2022 06:58 UTC'],
+            ['ros-system.internal', 'RHEL 8.4', '90%', '97%', '0.314', '1', 'Undersized', '03 Mar 2022 06:58 UTC']
+        ];
+
+        const testData = [
```

Comment:
`report_date` is not added under `testData`.

Reply:
@r14chandra removed it from tests for now, please refer to my comment [here](https://github.com/RedHatInsights/ros-frontend/pull/101#discussion_r836480902)